integration-tests/src/test/java/oracle/weblogic/kubernetes/ItHPA.java

```diff
@@ -0,0 +1,235 @@
+    assertNotNull(namespaces.get(1), "Namespace is null");
+    domainNamespace = namespaces.get(1);
+
+    // set the service account name for the operator
+    opServiceAccount = opNamespace + "-sa";
+
+    // install and verify operator with REST API
+    installAndVerifyOperator(opNamespace, opServiceAccount, true, 0, domainNamespace);
+
+    // create pull secrets for WebLogic image when running in non Kind Kubernetes cluster
+    // this secret is used only for non-kind cluster
+    createBaseRepoSecret(domainNamespace);
+
+    // create secret for admin credentials
+    logger.info("Create secret for admin credentials");
+    adminSecretName = "weblogic-credentials";
+    createSecretWithUsernamePassword(adminSecretName, domainNamespace,
+        ADMIN_USERNAME_DEFAULT, ADMIN_PASSWORD_DEFAULT);
+
+    // create encryption secret
+    logger.info("Create encryption secret");
+    encryptionSecretName = "encryptionsecret";
+    createSecretWithUsernamePassword(encryptionSecretName, domainNamespace,
+        "weblogicenc", "weblogicenc");
+
+    domain = createDomainResource(
+        domainUid,
+        domainNamespace,
+        MII_BASIC_IMAGE_NAME + ":" + MII_BASIC_IMAGE_TAG,
+        adminSecretName,
+        new String[]{TEST_IMAGES_REPO_SECRET_NAME},
+        encryptionSecretName
+    );
+
+    // create cluster resouce with limits and requests in serverPod
+    ClusterResource clusterResource =
+        createClusterResource(clusterResName, wlClusterName, domainNamespace, replicaCount);
+    clusterResource.getSpec()
+        .serverPod(new ServerPod().resources(
+            new V1ResourceRequirements()
+                .putLimitsItem("cpu", Quantity.fromString("2"))
+                .putLimitsItem("memory", Quantity.fromString("2Gi"))
+                .putRequestsItem("cpu", Quantity.fromString("250m"))
+                .putRequestsItem("memory", Quantity.fromString("768Mi"))));
+    logger.info("Creating cluster {0} in namespace {1}", clusterResName, domainNamespace);
+    createClusterAndVerify(clusterResource);
+
+    // set cluster references
+    domain.getSpec().withCluster(new V1LocalObjectReference().name(clusterResName));
+
+    // verify the domain custom resource is created
+    createDomainAndVerify(domain, domainNamespace);
+  }
+
+  @BeforeEach
+  public void beforeEach() {
+    // check admin server is up and running for domain1
+    checkPodReadyAndServiceExists(adminServerPodName, domainUid, domainNamespace);
+
+    // check managed servers are up and running for domain1
+    for (int i = 1; i <= replicaCount; i++) {
+      checkPodReadyAndServiceExists(managedServerPrefix + i, domainUid, domainNamespace);
+    }
+
+  }
+
+  @Test
+  void testHPAWithMetricsServer() {
+    // install metrics server
+    installMetricsServer();
+
+    // create hpa with autoscale
+    createHPA();
+
+    // create load on cpu and verify scale up/scale down
+    createLoadOnCpuAndVerifyAutoscaling();
```

Comment:
Can we make the inline code instead of function for better readability.

Reply:
isn’t it clear to read the code with methods rather than one linear big method?